### PR TITLE
Use new backend endpoint for checking handles availability

### PR DESCRIPTION
**Pull request by marcoconti83**
# Reason for this pull request
There is now a better endpoint on the backend to test for handles availability

### CODE REVIEW

File: Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift [hunk 1]
```diff
@@ -146,8 +146,11 @@ extension UserProfileRequestStrategy : ZMSingleRequestTranscoder {
             return ZMTransportRequest(path: "/self/handle", method: .methodPUT, payload: payload)
             
         case self.handleSuggestionSearchSync:
-            let handles = (self.userProfileUpdateStatus.suggestedHandlesToCheck ?? []).joined(separator: ",")
-            return ZMTransportRequest(path: "/users?handles=\(handles)", method: .methodGET, payload: nil)
+            let payload = [
+                    "handles" : self.userProfileUpdateStatus.suggestedHandlesToCheck ?? [],
```
Comment (daehn) [inline Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift]: Should we actually do a request if the array of suggestions is not there, I assume if this happens that something went wrong.

Comment (marcoconti83) [inline Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift]: True, will assert

Comment (marcoconti83) [inline Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift]: It's too late not to send a request at this point

**Comment by marcoconti83** [comment]
@daehn  Fixed

**Pull request merged**